src/v/storage/offset_to_filepos.cc

```diff
@@ -209,17 +209,25 @@ convert_end_offset_to_file_pos(
     auto res = co_await storage::transform_stream(
       reader_handle.take_stream(),
       std::move(ostr),
-      [off_end = end_inclusive, &fo, &ts, &offset_found](
+      [off_end = end_inclusive, &fo, &ts, &offset_found, &max_timestamp](
```

Comment:
nit: would be good to update first line of commit message to say that this commit is a fix (with tests) -- otherwise it looks like this commit was only adding a test.

Reply:
changed the commit message